### PR TITLE
update kcl to version 0.9.3

### PR DESCRIPTION
**Pull request by YvanDaSilva**
Updates kcl, kclvm and kclvm_cli to v0.9.3
Also addresses macos aarch64 platform support

## Description of changes

Changelog is here: https://github.com/kcl-lang/kcl/releases/tag/v0.9.3
and here: https://github.com/kcl-lang/cli/releases/tag/v0.9.3

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

Tested on latest darwin and nixos, respectively aarch64 and x86_64.

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

@Peefy @selfuryon

### CODE REVIEW

**Comment by YvanDaSilva** [comment]
@selfuryon You are marked as a reviewer, do you have time for it?

**Review by selfuryon — COMMENTED** [review]
I see there are already v0.9.3, maybe update all here to the latest version?

**Comment by YvanDaSilva** [comment]
@selfuryon I'll update to the latest now.

**Review by selfuryon — APPROVED** [review]
lgtm

File: pkgs/by-name/kc/kclvm_cli/package.nix [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
     repo = "kcl";
     rev = "v${version}";
-    hash = "sha256-ieGpuNkzT6AODZYUcEanb7Jpb+PXclnQ9KkdmlehK0o=";
+    hash = "sha256-nk5oJRTBRj0LE2URJqno8AoZ+/342C2tEt8d6k2MAc8=";
   };
 
   sourceRoot = "source/cli";
-  cargoLock.lockFile = ./Cargo.lock;
+
+  cargoLock = {
```
Comment (Aleksanaa) [inline pkgs/by-name/kc/kclvm_cli/package.nix:23]: You can delete `cargoLock` and regenerate `cargoHash` in this case.

**Review by Aleksanaa — COMMENTED** [review]
Please split into three commits, each with `<package attribute name>: <old version> -> <new version>`.

File: pkgs/by-name/kc/kclvm_cli/package.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/by-name/kc/kclvm_cli/package.nix:23]: Also delete `Cargo.lock`

Comment (YvanDaSilva) [reply pkgs/by-name/kc/kclvm_cli/package.nix]: @Aleksanaa Done :)

**Review by selfuryon — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/kc/kcl/package.nix [hunk 2]
```diff
@@ -1,44 +1,59 @@
...
 
   ldflags = [
     "-w -s"
     "-X=kcl-lang.io/cli/pkg/version.version=v${version}"
   ];
 
-  nativeBuildInputs = [ makeWrapper installShellFiles ];
-  buildInputs = [ kclvm kclvm_cli ];
+  nativeBuildInputs = [ makeWrapper installShellFiles ] ++ (
+    lib.optionals stdenv.isDarwin [ darwin.cctools ]
```
Comment (paparodeo) [inline pkgs/by-name/kc/kcl/package.nix:34]: `darwin.cctools` moved to `pkgs/by-name/cctools` in staging and staging-next so this is created eval errors.
https://gist.github.com/GrahamcOfBorg/cc66a857ef768344f881acbbaacc32bf which is not a big deal, i'm looking at fixing it. however, isn't cctools already part of the stdenv? Perhaps i'm missing something but it seems like all of the `darwin.cctools` can just be removed and not effect the build.

Comment (paparodeo) [reply pkgs/by-name/kc/kcl/package.nix:34]: builds fine on x64 darwin removing the cctools from inputs

https://github.com/NixOS/nixpkgs/pull/330118

Comment (YvanDaSilva) [reply pkgs/by-name/kc/kcl/package.nix:34]: @paparodeo Let me test it, I'll be right back at you, if that's the case. I'll open a separate PR.

**Pull request merged**